soh/soh/Enhancements/randomizer/3drando/location_access/locacc_ganons_castle.cpp

```diff
@@ -130,7 +130,7 @@ void AreaTable_Init_GanonsCastle() {
   areaTable[GANONS_CASTLE_MQ_LOBBY] = Area("Ganon's Castle MQ Lobby", "Ganons Castle", GANONS_CASTLE, NO_DAY_NIGHT_CYCLE, {}, {}, {
                   //Exits
                   Entrance(GANONS_CASTLE_ENTRYWAY, {[]{return true;}}),
-                  Entrance(GANONS_CASTLE_MQ_MAIN,  {[]{return (IsAdult || (HasExplosives || ((Nuts || Boomerang) && (Sticks || KokiriSword))));}}),
+                  Entrance(GANONS_CASTLE_MQ_MAIN,  {[]{return (CanUse(MASTER_SWORD) || CanUse(BIGGORON_SWORD)) || ((HasExplosives || Nuts || CanUse(BOOMERANG)) && CanJumpslash);}}),
```

Comment:
i think the has explosives needs to be in the first group of ORs? unless the previous logic was wrong

Reply:
It is different now. The logic wasn't necessarily wrong, but I 3ds rando has since changed it to this, and n64 rando has similar-ish logic.

I believe the thinking is that although you could technically kill all the enemies with just explosives alone, you'd have to be pretty strategic with it (hitting multiple enemies at the same time with one bomb, getting the bomb drops from armos if you have a bomb bag/bombchus in logic). So I think they start requiring a second item so it didn't depend on you having a "larger enough" bomb bag or precise attacks.

Considering we also have an enhancement to disable drops, I think its best to not rely on drops mid-battle.